Travis improvements

pnorman opened the pull request:
- Use `after_failure` to greatly simplify logic
- Use containterized builds for faster builds

(still needs "testing" on travis)

pnorman committed:
Use travis after_failure to simplify logic

pnorman committed:
Use container-based travis infrastructure

pnorman commented:
This is executing the find ... gdb line, but a dump doesn't seem to have been generated

jfirebaugh commented:
The `after_failure` use looks like a good idea.

Containerized builds can't currently generate core dumps -- see https://github.com/travis-ci/travis-ci/issues/3754#issuecomment-95629153.

pnorman commented:
> Containerized builds can't currently generate core dumps -- see travis-ci/travis-ci#3754 (comment).

ah. That'd also explain the problems I was having

pnorman committed:
Set sudo:true

Until travis-ci/travis-ci#3754 is fixed, sudo is needed, and it adds
clarity to explicitly set it.

pnorman commented:
The travis build is now a "success"

```
$ for i in $(find ./ -maxdepth 1 -name 'core*' -print); do gdb $(pwd)/test core* -ex "thread apply all bt" -ex "set pagination 0" -batch; done;
[New LWP 3153]
warning: Can't read pathname for load map: Input/output error.
Core was generated by `./test'.
Program terminated with signal 11, Segmentation fault.
#0  0x00007f77e454b02c in free () from /lib/x86_64-linux-gnu/libc.so.6
Thread 1 (LWP 3153):
#0  0x00007f77e454b02c in free () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x00000000004005d6 in main () at might_crash.cpp:6
```

pnorman commented:
Closing as abandoned
